nx-X11/programs/Xserver/hw/nxagent/Options.h

```
@@ -384,6 +384,9 @@ typedef struct _AgentOptions
 
   /*
    * Inhibit some XKEYBOARD requests.
+   * 0 = allow all XKEYBOARD requests
+   * 1 = forbid all XKEYBOARD requests if keyboard=clone or keyboard=query
```

Also wrong, this is actually the default - regardless of the `keyboard` setting.

---

I wonder if `InhibitXkb` is actually doing what it's supposed to do? We should see denied change requests a lot more often if it were working correctly.

---

Ah, and it makes sense: this the XKB extension is enabled when a client (re-)connects (and our first connection is also a reconnection for that matter), this will always be the case and disabling the XKB extension pretty much useless.

---

Again I am not sure what you mean. Have look at nxagentInitXkbWrapper().

If `InihibitXkb` is `0` no own filter wrapper is installed. Which effectively means that all requests are accepted.

If `InihibitXkb` is 1 (the default) all _writing_ XKEYBOARD requests are silently ignored, the others are accepted. 

If `InihibitXkb` is 2 the behaviour is like 1 but only if neither query nor clone are used. If so, the behaviour is like 0.

So my description is wrong, I have improved that.

However, I have just tested again and found that 1 is only working for clone and query (as I wrote in the code) despite being the default. This needs further investigation.